crates/core/src/vm.rs

```diff
@@ -268,6 +267,18 @@ fn get_table<'a>(
     })
 }
 
+fn build_source_query<'a, I: 'a + Iterator<Item = RelValue<'a>>>(
```

Comment:
I find this name unhelpful; why does a "source query" need to be a `MemTable`? Surely compound queries are allowed to have `DbTable` sources.